contrib/mesos/pkg/queue/historical.go

```diff
@@ -377,6 +377,19 @@ func (f *HistoricalFIFO) merge(id string, obj UniqueCopyable) (notifications []E
 	return
 }
 
+// Resync will touch all objects to put them into the processing queue
+func (f *HistoricalFIFO) Resync() error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	for id, entry := range f.items {
+		if entry.Is(ADD_EVENT|UPDATE_EVENT) {
+			f.queue = append(f.queue, id)
+		}
+	}
+	f.cond.Broadcast()
+	return nil
```

Comment:
So basically this is "SYNC" operation. E.g. in DeltaFifo we have a dedicated event type for it:
https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/delta_fifo.go#L499

I'm not sure if you need it or not in your code.